x-pack/plugin/old-lucene-versions/src/main/java/org/elasticsearch/xpack/lucene/bwc/codecs/LegacyAdaptingPerFieldPostingsFormat.java

```diff
@@ -68,39 +65,19 @@ public abstract class LegacyAdaptingPerFieldPostingsFormat extends PostingsForma
     public static final String PER_FIELD_SUFFIX_KEY = PerFieldPostingsFormat.class.getSimpleName() + ".suffix";
 
     /** Sole constructor. */
-    protected LegacyAdaptingPerFieldPostingsFormat() {
+    public LegacyAdaptingPerFieldPostingsFormat() {
         super(PER_FIELD_NAME);
     }
 
     static String getSuffix(String formatName, String suffix) {
         return formatName + "_" + suffix;
     }
 
-    protected PostingsFormat getPostingsFormat(String formatName) {
-        throw new IllegalArgumentException(formatName);
-    }
-
-    private class FieldsWriter extends FieldsConsumer {
```

Comment:
there is no need to have a FieldsWriter that throws exceptions, we can simply throw exception directly when a fields writer is requested, as that's only used for writing.